### PR TITLE
Notification list

### PR DESCRIPTION
**Pull request by Elly-G**
add notification list

### CODE REVIEW

File: intellst-api/src/DTO/IdentifiedCaseDTO.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 use DateTime;
 use JMS\Serializer\Annotation as Serializer;
+use phpDocumentor\Reflection\Types\Integer;
```
Comment (andreikojusko) [inline intellst-api/src/DTO/IdentifiedCaseDTO.php]: not needed

File: intellst-api/src/Entity/IdentifiedCase.php [hunk 2]
```diff
@@ -7,6 +7,7 @@
 use DateTimeInterface;
 use Doctrine\ORM\Mapping as ORM;
 use Symfony\Component\Validator\Constraints as Assert;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
```
Comment (andreikojusko) [inline intellst-api/src/Entity/IdentifiedCase.php]: is it used?

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 3]
```diff
@@ -33,22 +33,79 @@ public function removeCases()
...
-    /*
-    public function findByExampleField($value)
+    public function showNewIdentifiedCase(int $day, float $temperature)
     {
-        return $this->createQueryBuilder('i')
-            ->andWhere('i.exampleField = :val')
-            ->setParameter('val', $value)
-            ->orderBy('i.id', 'ASC')
-            ->setMaxResults(10)
+        $period = strval($day);
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: not needed

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 4]
```diff
@@ -33,22 +33,79 @@ public function removeCases()
             ->execute();
     }
 
-    // /**
-    //  * @return IdentifiedCase[] Returns an array of IdentifiedCase objects
-    //  */
-    /*
-    public function findByExampleField($value)
+    public function showNewIdentifiedCase(int $day, float $temperature)
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: why "show"? isn't it "get"? same in other "show" methods

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 5]
```diff
@@ -33,22 +33,79 @@ public function removeCases()
...
             ->getQuery()
-            ->getResult()
-        ;
+            ->execute();
     }
-    */
+
+    public function returnAttempt(int $day, float $temperature, string $uuid)
+    {
+        $period = strval($day);
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: not needed

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 6]
```diff
@@ -33,22 +33,79 @@ public function removeCases()
...
+            ->getQuery()
+            ->execute();
+    }
+
+    public function allowEntrance(string $uuid)
+    {
+        $data = new \DateTime();
+        $data->modify('-1 day');
+        return $this->createQueryBuilder('IdentifiedCase')
+            ->Where('IdentifiedCase.allowEntrance > :date')
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: "where" (lowercase w)

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 7]
```diff
@@ -33,22 +33,79 @@ public function removeCases()
...
+            ->setParameter(':date', $data)
+            ->andWhere('IdentifiedCase.uuid = :uuid')
+            ->setParameter(':uuid', $uuid)
+            ->getQuery()
+            ->execute();
+    }
+
+    public function allowEntrance(string $uuid)
+    {
+        $data = new \DateTime();
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: `$date`

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 8]
```diff
@@ -33,22 +33,79 @@ public function removeCases()
...
+//    {
+//        return $this->createQueryBuilder('i')
+//            ->andWhere('i.temperature = :temperature')
+//            ->setParameter('temperature', $value)
+//            ->orderBy('i.id', 'ASC')
+//            ->setMaxResults(10)
+//            ->getQuery()
+//            ->getResult()
+//        ;
+//    }
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: is this code needed?

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 9]
```diff
@@ -33,22 +33,79 @@ public function removeCases()
...
+        return $this->createQueryBuilder('IdentifiedCase')
+            ->Where('IdentifiedCase.firstDate > :date')
+            ->setParameter(':date', $data)
+            ->andWhere('IdentifiedCase.uuid = :uuid')
+            ->setParameter(':uuid', $uuid)
+            ->getQuery()
+            ->execute();
+    }
+
+    public function allowEntrance(string $uuid)
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: isEntranceAllowed

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 10]
```diff
@@ -33,22 +33,79 @@ public function removeCases()
...
+    {
+        $data = new \DateTime();
+        $data->modify('-1 day');
+        return $this->createQueryBuilder('IdentifiedCase')
+            ->Where('IdentifiedCase.allowEntrance > :date')
+            ->setParameter(':date', $data)
+            ->andWhere('IdentifiedCase.uuid = :uuid')
+            ->setParameter(':uuid', $uuid)
+            ->getQuery()
+            ->execute();
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: why "execute" and not "getResult"?

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 11]
```diff
@@ -33,22 +33,79 @@ public function removeCases()
...
+
+    public function showReturnAttempt(string $uuid, $data)
+    {
+        return $this->createQueryBuilder('IdentifiedCase')
+            ->Where('IdentifiedCase.firstDate > :date')
+            ->setParameter(':date', $data)
+            ->andWhere('IdentifiedCase.uuid = :uuid')
+            ->setParameter(':uuid', $uuid)
+            ->getQuery()
+            ->execute();
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: why "execute" and not "getResult"?

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 12]
```diff
@@ -33,22 +33,79 @@ public function removeCases()
...
+
+        return $this->createQueryBuilder('IdentifiedCase')
+            ->Where('IdentifiedCase.firstDate > :date')
+            ->setParameter(':date', $date)
+            ->andWhere('IdentifiedCase.temperature > :temperature')
+            ->setParameter(':temperature', $temperature)
+            ->andWhere('IdentifiedCase.uuid = :uuid')
+            ->setParameter(':uuid', $uuid)
+            ->getQuery()
+            ->execute();
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: why "execute" and not "getResult"?

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 13]
```diff
@@ -33,22 +33,79 @@ public function removeCases()
...
+            ->andWhere('IdentifiedCase.temperature > :temperature')
+            ->setParameter(':temperature', $temperature)
             ->getQuery()
-            ->getResult()
-        ;
+            ->execute();
     }
-    */
+
+    public function returnAttempt(int $day, float $temperature, string $uuid)
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: probably, "getReturnAttempts"

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 14]
```diff
@@ -33,22 +33,79 @@ public function removeCases()
...
+            ->andWhere('IdentifiedCase.uuid = :uuid')
+            ->setParameter(':uuid', $uuid)
+            ->getQuery()
+            ->execute();
+    }
+
+    public function showReturnAttempt(string $uuid, $data)
+    {
+        return $this->createQueryBuilder('IdentifiedCase')
+            ->Where('IdentifiedCase.firstDate > :date')
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: lowercase "w"

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 15]
```diff
@@ -33,22 +33,79 @@ public function removeCases()
...
+            ->setParameter(':date', $date)
+            ->andWhere('IdentifiedCase.temperature > :temperature')
+            ->setParameter(':temperature', $temperature)
+            ->andWhere('IdentifiedCase.uuid = :uuid')
+            ->setParameter(':uuid', $uuid)
+            ->getQuery()
+            ->execute();
+    }
+
+    public function showReturnAttempt(string $uuid, $data)
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: $date

**Review by andreikojusko — CHANGES_REQUESTED** [review]

File: intellst-api/src/Entity/Enterprise.php [hunk 16]
```diff
@@ -51,9 +51,15 @@ class Enterprise
      */
     private int $restrictionPeriod;
 
+    /**
+     * @ORM\OneToMany(targetEntity=IdentifiedCase::class, mappedBy="enterprise")
+     */
+    private $identifiedCase;
```
Comment (andreikojusko) [inline intellst-api/src/Entity/Enterprise.php]: $identifiedCases (multiple)

File: intellst-api/src/DTO/IdentifiedCaseDTO.php [hunk 17]
```diff
@@ -44,4 +52,15 @@ class IdentifiedCaseDTO
...
+    /**
+     * Enterprise IdentifiedCase
+     * @var integer
+     * @Assert\NotNull
+     * @Serializer\Type("integer")
+     * @Serializer\Expose()
+     * @Serializer\SerializedName("enterprise")
+     */
+    public $enterprise;
+
```
Comment (andreikojusko) [inline intellst-api/src/DTO/IdentifiedCaseDTO.php]: empty line not needed

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 18]
```diff
@@ -27,38 +27,64 @@ public function removeCases()
...
+
+        return $this->createQueryBuilder('IdentifiedCase')
+            ->where('IdentifiedCase.firstDate > :date')
+            ->setParameter(':date', $date)
+            ->andWhere('IdentifiedCase.temperature > :temperature')
+            ->setParameter(':temperature', $temperature)
             ->getQuery()
-            ->getResult()
-        ;
+            ->execute();
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: why "execute" and not "getResult"?

File: intellst-api/src/Services/UserHandler.php [hunk 19]
```diff
@@ -40,20 +41,17 @@ public function __construct(
...
-    {
+    ) {
         $this->em = $em;
         $this->transformer = $transformer;
         $this->userRepository = $this->em->getRepository(User::class);
         $this->user = $tokenStorage->getToken()->getUser();
     }
 
-    public function getList(): array
+    public function getList(): UserDTO
```
Comment (andreikojusko) [inline intellst-api/src/Services/UserHandler.php]: why getList returns single user? method should be named `getCurrentUser`

File: intellst-api/src/Transformer/EnterpriseTransformer.php [hunk 20]
```diff
@@ -15,4 +15,13 @@ public function transformDTOToEntity(EnterpriseDTO $dto, Enterprise $enterprise)
...
 
+    public function transformEntityToDTO(Enterprise $enterprise): EnterpriseDTO
+    {
+        $enterpriseDTO = new EnterpriseDTO();
+        $enterpriseDTO->temperature = $enterprise->getTemperature();
+        $enterpriseDTO->restrictionPeriod = $enterprise->getRestrictionPeriod();
+
+        return $enterpriseDTO;
+    }
+
```
Comment (andreikojusko) [inline intellst-api/src/Transformer/EnterpriseTransformer.php]: extra line

File: intellst-api/src/Transformer/IdentifiedCaseTransformer.php [hunk 21]
```diff
@@ -4,18 +4,35 @@
...
 class IdentifiedCaseTransformer
 {
+    /**
+     * @var EntityManagerInterface
+     */
+    private $em;
+
+    public function __construct(EntityManagerInterface $em)
+    {
+        $this->em = $em;
```
Comment (andreikojusko) [inline intellst-api/src/Transformer/IdentifiedCaseTransformer.php:16]: looks like `$this->em` is never used

File: intellst-api/src/Transformer/IdentifiedCaseTransformer.php [hunk 22]
```diff
@@ -4,18 +4,35 @@
...
         $identifiedCase->setTemperature($dto->temperature);
         $identifiedCase->setDatePhoto(new \DateTime());
         $identifiedCase->setFirstDate(new \DateTime());
-
+        $data = new \DateTime();
+        $data->modify('-1 day');
+        $identifiedCase->setAllowEntrance($data);
+        $enterprise = $this->em->getRepository(Enterprise::class)->find($dto->enterprise);
+        $identifiedCase->setEnterprise($enterprise);
         return $identifiedCase;
```
Comment (andreikojusko) [inline intellst-api/src/Transformer/IdentifiedCaseTransformer.php:33]: blank line before return please

File: intellst-api/src/Transformer/UserTransformer.php [hunk 23]
```diff
@@ -14,7 +14,7 @@ public function transformEntityToDTO(User $user): UserDTO
         $userDTO->firstname = $user->getFirstname();
         $userDTO->lastname = $user->getLastname();
         $userDTO->email = $user->getEmail();
-
+        $userDTO->enterprise = $user->getEnterprise()->getId();
         return $userDTO;
```
Comment (andreikojusko) [inline intellst-api/src/Transformer/UserTransformer.php:19]: blank line before return please

File: intellst-api/src/Services/IdentifiedCaseHandler.php [hunk 24]
```diff
@@ -81,4 +102,92 @@ public function updateIdentifiedCaseAllowEntrance(AllowEntranceDTO $dto, Identif
 
         return $errors;
     }
+
+    public function getIdentifiedCase(): array
```
Comment (andreikojusko) [inline intellst-api/src/Services/IdentifiedCaseHandler.php]: `getIdentifiedCases` (multiple)

File: intellst-api/src/Services/IdentifiedCaseHandler.php [hunk 25]
```diff
@@ -81,4 +102,92 @@ public function updateIdentifiedCaseAllowEntrance(AllowEntranceDTO $dto, Identif
...
+                $identifiedCaseDTO->firstDate
+            );
+            if (!empty($returnAttempt)) {
+                foreach ($returnAttempt as $case) {
+                    $caseDTO = $this->identifiedCaseTransformer->transformEntityToDTO($case);
+                    $arr[] = $caseDTO;
+                }
+            }
+        }
+        return $arr;
```
Comment (andreikojusko) [inline intellst-api/src/Services/IdentifiedCaseHandler.php:165]: blank line before return

File: intellst-api/src/Services/IdentifiedCaseHandler.php [hunk 26]
```diff
@@ -81,4 +102,92 @@ public function updateIdentifiedCaseAllowEntrance(AllowEntranceDTO $dto, Identif
...
+
+    private function getEnterprise(IdentifiedCaseDTO $dto): EnterpriseDTO
+    {
+        $enterprise = $this->enterpriseRepository->find($dto->enterprise);
+        $enterpriseDTO = $this->enterpriseTransformer->transformEntityToDTO($enterprise);
+
+        return $enterpriseDTO;
+    }
+
+    private function returnUserEnterprise(UserDTO $dto): EnterpriseDTO
```
Comment (andreikojusko) [inline intellst-api/src/Services/IdentifiedCaseHandler.php]: `getUserEnterprise`
Also, why have 2 methods? Isn't it simplied to have 1 method that return EnterpriseDTO by enterprise ID? You can have just `getEnterprise(int $id)`

File: intellst-api/src/Services/IdentifiedCaseHandler.php [hunk 27]
```diff
@@ -81,4 +102,92 @@ public function updateIdentifiedCaseAllowEntrance(AllowEntranceDTO $dto, Identif
 
         return $errors;
     }
+
+    public function getIdentifiedCase(): array
+    {
+        $user = $this->userHandler->getList();
+        $temperature = $this->returnUserEnterprise($user)->temperature;
+        $data = $this->returnUserEnterprise($user)->restrictionPeriod;
```
Comment (andreikojusko) [inline intellst-api/src/Services/IdentifiedCaseHandler.php]: not need to make 2 requests to `returnUserEnterprise`
```
$dto = $this->returnUserEnterprise($user);
```
then use `$dto->temperature`

Comment (andreikojusko) [inline intellst-api/src/Services/IdentifiedCaseHandler.php]: `$data` isn't a good name. If it is a number of days, then `$days` if it is DateTime, then `$date`

File: intellst-api/src/Services/IdentifiedCaseHandler.php [hunk 28]
```diff
@@ -81,4 +102,92 @@ public function updateIdentifiedCaseAllowEntrance(AllowEntranceDTO $dto, Identif
...
+        $users = $this->identifiedCaseRepository->getNewIdentifiedCase($data, $temperature);
```
Comment (andreikojusko) [inline intellst-api/src/Services/IdentifiedCaseHandler.php]: `$users` or `$cases`?

File: intellst-api/src/Services/IdentifiedCaseHandler.php [hunk 29]
```diff
@@ -81,4 +102,92 @@ public function updateIdentifiedCaseAllowEntrance(AllowEntranceDTO $dto, Identif
...
+            $arr[] = $identifiedCaseDTO;
+        }
+
+        return $arr;
+    }
+
+    public function getReturnAttempts(IdentifiedCaseDTO $dto): bool
+    {
+        $temperature = $this->getEnterprise($dto)->temperature;
+        $day = $this->getEnterprise($dto)->restrictionPeriod;
```
Comment (andreikojusko) [inline intellst-api/src/Services/IdentifiedCaseHandler.php]: don't call `getEnterprise` twice

File: intellst-api/src/Services/IdentifiedCaseHandler.php [hunk 30]
```diff
@@ -81,4 +102,92 @@ public function updateIdentifiedCaseAllowEntrance(AllowEntranceDTO $dto, Identif
...
+        $arr = [];
+        foreach ($users as $identifiedCase) {
+            $identifiedCaseDTO = $this->identifiedCaseTransformer->transformEntityToDTO($identifiedCase);
+            $arr[] = $identifiedCaseDTO;
+        }
+
+        return $arr;
+    }
+
+    public function getReturnAttempts(IdentifiedCaseDTO $dto): bool
```
Comment (andreikojusko) [inline intellst-api/src/Services/IdentifiedCaseHandler.php]: why `getReturnAttempts` if it returns a boolean? `it should be `isReturnAttempt`

File: intellst-api/src/Services/IdentifiedCaseHandler.php [hunk 31]
```diff
@@ -81,4 +102,92 @@ public function updateIdentifiedCaseAllowEntrance(AllowEntranceDTO $dto, Identif
...
+        $arr = [];
+        foreach ($list as $identifiedCase) {
+            $identifiedCaseDTO = $this->identifiedCaseTransformer->transformEntityToDTO($identifiedCase);
+            $arr[] = $identifiedCaseDTO;
+        }
+        if (!empty($arr)) {
+            return true;
+        } else {
+            return false;
+        }
```
Comment (andreikojusko) [inline intellst-api/src/Services/IdentifiedCaseHandler.php]: why do you need all this code? no need in transformations at all
```
$list = $this->identifiedCaseRepository->getReturnAttempts($day, $temperature, $uuid);

return !(empty($list));
```

File: intellst-api/src/Services/IdentifiedCaseHandler.php [hunk 32]
```diff
@@ -81,4 +102,92 @@ public function updateIdentifiedCaseAllowEntrance(AllowEntranceDTO $dto, Identif
...
+    }
+
+    public function entranceAllowed(IdentifiedCaseDTO $dto): bool
```
Comment (andreikojusko) [inline intellst-api/src/Services/IdentifiedCaseHandler.php]: `isEntranceAllowed`

File: intellst-api/src/Services/IdentifiedCaseHandler.php [hunk 33]
```diff
@@ -81,4 +102,92 @@ public function updateIdentifiedCaseAllowEntrance(AllowEntranceDTO $dto, Identif
...
+        $arr = [];
+        foreach ($list as $identifiedCase) {
+            $identifiedCaseDTO = $this->identifiedCaseTransformer->transformEntityToDTO($identifiedCase);
+            $arr[] = $identifiedCaseDTO;
+        }
+        if (!empty($arr)) {
+            return true;
+        } else {
+            return false;
+        }
```
Comment (andreikojusko) [inline intellst-api/src/Services/IdentifiedCaseHandler.php]:
```
$list = $this->identifiedCaseRepository->entranceAllowed($uuid);

return !(empty($list));
```

File: intellst-api/src/Services/IdentifiedCaseHandler.php [hunk 34]
```diff
@@ -81,4 +102,92 @@ public function updateIdentifiedCaseAllowEntrance(AllowEntranceDTO $dto, Identif
...
+
+    private function returnUserEnterprise(UserDTO $dto): EnterpriseDTO
+    {
+        $enterprise = $this->enterpriseRepository->find($dto->enterprise);
+        $enterpriseDTO = $this->enterpriseTransformer->transformEntityToDTO($enterprise);
+
+        return $enterpriseDTO;
+    }
+
+    public function getReturnAttempt(): array
```
Comment (andreikojusko) [inline intellst-api/src/Services/IdentifiedCaseHandler.php]: please review naming, calls, logic as in comments for other methods

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 35]
```diff
@@ -27,38 +27,64 @@ public function removeCases()
...
+            ->setParameter(':date', $date)
+            ->andWhere('IdentifiedCase.temperature > :temperature')
+            ->setParameter(':temperature', $temperature)
+            ->andWhere('IdentifiedCase.uuid = :uuid')
+            ->setParameter(':uuid', $uuid)
+            ->getQuery()
+            ->execute();
+    }
+
+    public function getReturnAttempt(string $uuid, $data)
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: `getReturnAttempts` (multiple), return type, `$date`

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 36]
```diff
@@ -27,38 +27,64 @@ public function removeCases()
...
             ->getQuery()
-            ->getResult()
-        ;
+            ->execute();
     }
-    */
 
-    /*
-    public function findOneBySomeField($value): ?IdentifiedCase
+    public function getReturnAttempts(int $day, float $temperature, string $uuid)
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: missing return type

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 37]
```diff
@@ -27,38 +27,64 @@ public function removeCases()
...
+            ->andWhere('IdentifiedCase.uuid = :uuid')
+            ->setParameter(':uuid', $uuid)
+            ->getQuery()
+            ->getResult();
+    }
+
+    public function entranceAllowed(string $uuid)
+    {
+        $date = new \DateTime();
+        $date->modify('-1 day');
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php:81]: why 1 day???

File: intellst-api/src/Repository/IdentifiedCaseRepository.php [hunk 38]
```diff
@@ -27,38 +27,64 @@ public function removeCases()
...
+        return $this->createQueryBuilder('IdentifiedCase')
+            ->where('IdentifiedCase.firstDate > :date')
+            ->setParameter(':date', $data)
+            ->andWhere('IdentifiedCase.uuid = :uuid')
+            ->setParameter(':uuid', $uuid)
+            ->getQuery()
+            ->getResult();
+    }
+
+    public function entranceAllowed(string $uuid)
```
Comment (andreikojusko) [inline intellst-api/src/Repository/IdentifiedCaseRepository.php]: name isn't good, but I can't suggest better name as I don't have any idea what should this method do, also missing return type

**Review by andreikojusko — CHANGES_REQUESTED** [review]

File: intellst-api/src/Transformer/IdentifiedCaseTransformer.php (hunk 21 above)
Comment (Elly-G) [reply intellst-api/src/Transformer/IdentifiedCaseTransformer.php:16]: is used in transformDTOToEntity 
` $enterprise = $this->em->getRepository(Enterprise::class)->find($dto->enterprise); `

**Pull request merged**